fuse_models/src/acceleration_2d.cpp

```diff
@@ -73,8 +76,8 @@ void Acceleration2D::onStart()
 {
   if (!params_.indices.empty())
   {
-    subscriber_ =
-      node_handle_.subscribe(ros::names::resolve(params_.topic), params_.queue_size, &Acceleration2D::process, this);
+    subscriber_ = node_handle_.subscribe(ros::names::resolve(params_.topic), params_.queue_size,
```

Comment:
Interesting. I wonder if a `ThrottledSubscriber` would result in more idiomatic usage?
Something like:
```
using Callback = std::function<void(const MessageConstPtr&)>;
ThrottledSubscriber(
  const ros::NodeHandle& node_handle,
  const std::string& topic_name,
  uint32_t queue_size,
  const ros::Duration& throttle_period,
  Callback keep_callback,
  Callback drop_callback = nullptr,
  const ros::TransportHints& transport_hints = ros::TransportHints());
```
And then when you subscribe, it looks "more normal":
```
subscriber_ = ThrottledSubscriber(node_handle_, ros::names::resolve(params_.topic), params_.queue_size, params_.throttle_period, boost::bind(&Acceleration2D::process, this, _1));
```

Just a thought. The throttled callback works just fine as well.

Reply:
The main reason I took the throttle callback path was that the ROS subscriber API supports multiple different input types, so I was afraid that would be a limitation. Besides that, a throttle callback is a more generic component that could also be used without ROS subscribers.